redaxo/src/core/lib/base/instance_list_pool_trait.php

```diff
@@ -49,6 +50,12 @@ protected static function hasInstanceList($key)
      * @param callable $createListCallback  Callback, will be called to create the list of instance keys
      *
      * @return array
+     *
+     * @template T as object
+     * @template TKey
+     * @psalm-param callable(TKey...):?T $getInstanceCallback
+     * @psalm-param callable(mixed...):TKey[] $createListCallback
+     * @psalm-return T[]
```

Comment:
Ich hab hier über Templates die Beziehungen zwischen den Parametern und Return-Types noch genauer spezifiziert. Allerdings klappt es noch nicht: https://github.com/vimeo/psalm/issues/3752

Reply:
Ohh super. Hab auch schon einiges probiert und mich gewundert wo es noch klemmt ;-)

Reply:
(Wobei die Beziehung bzgl. TKey so doch nicht stimmte; ist komplizierter, und habe ich zurück in mixed geändert.)

Reply:
Das Problem in psalm besteht zwar noch, aber ich habe es nun bei uns anders gelöst.